standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -1491,6 +1505,22 @@ boolean listPartitionsByExpr(String catName, String db_name, String tbl_name, by
                                        List<String> part_names)
       throws NoSuchObjectException, MetaException, TException;
 
+    /**
+     * Get partitions by a list of partition names.
+     * @param catName catalog name
+     * @param db_name database name
+     * @param tbl_name table name
+     * @param part_names list of partition names
+     * @param getColStats if true, column statistics is added to the Partition objects
+     * @return list of Partition objects
+     * @throws NoSuchObjectException No such partitions
+     * @throws MetaException error accessing the RDBMS.
+     * @throws TException thrift transport error
+     */
+    List<Partition> getPartitionsByNames(String catName, String db_name, String tbl_name,
```

Comment:
That's an existing problem. catName and db_name inconsistency existed even before I added getColStats, which aligns more with catName. If we try to make conventions uniform, it will cause unnecessary diffs which will affect further back-port and merged difficult. Generally, this file hasn't been following uniform conventions for all its APIs, so making one of them uniform wouldn't help much. So, I would leave it like that in this patch.

Reply:
ok